src/Arcus.EventGrid/Parsers/EventParser.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Arcus.EventGrid.Contracts;
+using GuardNet;
+using Newtonsoft.Json.Linq;
+
+namespace Arcus.EventGrid.Parsers 
+{
+    /// <summary>
+    /// Provides event parsing of raw JSON payloads to an abstracted <see cref="Event"/>.
+    /// </summary>
+    public static class EventParser
+    {
+        /// <summary>
+        /// Loads a raw JSON payload into an abstracted event.
+        /// </summary>
+        /// <param name="rawJson">The raw JSON payload, representing an event that can be handled by EventGrid.</param>
+        public static EventGridEventBatch<Event> Parse(byte[] rawJson)
+        {
+            Guard.NotNull(rawJson, nameof(rawJson), "Cannot parse a 'null' series of bytes raw JSON payload to an abstracted event");
+            Guard.NotLessThanOrEqualTo(rawJson.Length, threshold: 0, "Cannot parse a series of bytes of a length <= 0 to an abstracted event");
+
+            EventGridEventBatch<Event> eventBatch = Parse(rawJson, sessionId: Guid.NewGuid().ToString());
+            return eventBatch;
+        }
+
+        /// <summary>
+        /// Loads a raw JSON payload into an abstracted event with a specific <paramref name="sessionId"/>.
+        /// </summary>
+        /// <param name="rawJson">The raw JSON payload, representing an event that can be handled by EventGrid.</param>
+        /// <param name="sessionId">The reference ID for this event parsing session.</param>
+        public static EventGridEventBatch<Event> Parse(byte[] rawJson, string sessionId)
+        {
+            Guard.NotNull(rawJson, nameof(rawJson), "Cannot parse a 'null' series of bytes raw JSON payload to an abstracted event");
+            Guard.NotLessThanOrEqualTo(rawJson.Length, threshold: 0, "Cannot parse a series of bytes of a length <= 0 to an abstracted event");
+
+            // TODO: CloudEvents SDK can also parse byte arrays directly, but it can also parse directly JObjects (which we use now). Should we use both?
+            string json = Encoding.UTF8.GetString(rawJson);
```

Comment:
Why are we doing this here, but not for `Parse(byte[] rawJson)`

Reply:
I removed the TODO bc we should support array of events too.

Reply:
I think you missread? I happens for both.

Reply:
I have looked at the SDK implementation, they use the same approach when decoding bytes. We use instead the JObject decoding because we can then both parse array of events and a single event.